File: app/models/selected_seat.py
from app.extensions import db

class SelectedSeat(db.Model):
    __tablename__ = 'selected_seat'

    # 선택 좌석 PK
    selected_seat_id = db.Column(
        db.BigInteger,
        primary_key=True,
        autoincrement=True
    )

    # 회원 번호
    user_no = db.Column(
        db.BigInteger,
        db.ForeignKey('user.user_no'),
        nullable=False
    )

    # 좌석 ID
    seat_id = db.Column(
        db.String(20),
        db.ForeignKey('seat.seat_id'),
        nullable=False
    )

    # 좌석 상태
    # 0: 미선택
    # 1: 다른 유저 선택/결제중
    # 2: 선택완료
    seat_state = db.Column(
        db.SmallInteger,
        nullable=False,
        default=0
    )

    # 좌석 가격
    price = db.Column(
        db.BigInteger,
        nullable=False
    )

    reservation = db.relationship('Reservation', backref='selected_seat', lazy=True)

File: src/main/java/com/ticketflow/entity/SelectedSeat.java
package com.ticketflow.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "selected_seat")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SelectedSeat {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "selected_seat_id")
    private Long selectedSeatId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_no", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "seat_id", nullable = false)
    private Seat seat;

    // 0: 미선택, 1: 선택/결제중, 2: 선택완료
    @Column(name = "seat_state", nullable = false)
    @Builder.Default
    private Short seatState = 0;

    @Column(name = "price", nullable = false)
    private Long price;

    @OneToMany(mappedBy = "selectedSeat", fetch = FetchType.LAZY)
    @Builder.Default
    private List<Reservation> reservations = new ArrayList<>();
}
